docs: :construction: move app-sdk docs (#865)
saleor/saleor-docs: docs/developer/extending/apps/developing-apps/app-sdk/protected-views.md

> # App-protected views

Reviewer: **Suggestion[^1][^2]**:
Maybe a "Protecting the App dashboard interface" would be a better name?

[^1]: ✅ Not blocking the review. 
[^2]: It might have an impact on the submitted code. I've probably had a similar issue with the chosen approach. If you're sure that's overkill or does not apply to this PR, let me know and resolve the request.

Author: This one displays in the sidebar, I would rather keep it a bit shorter.

> Most likely, at least some screens of the Saleor App are meant to be visible only within the Dashboard "Apps" section.
> 
> SDK provides a Higher Order Component that can be used to wrap Next.js pages and perform basic protection checks

Reviewer: ```suggestion
SDK provides a Higher Order Component that can be used to wrap Next.js pages and perform basic protection checks. With it, you'll be able to make pages visible only when accessed from within Saleor Dashboard.
```

WDYT?
